src/amici_interface_matlab.cpp

```diff
@@ -468,17 +264,274 @@ ExpData *expDataFromMatlabCall(const mxArray *prhs[], const UserData *udata,
     }
     return edata;
 }
-    
-    /** conversion from double to int with checking for loss of data
-     *  @param x input
-     *  @return int_x casted value
-     */
-    int dbl2int(const double x){
-        if((std::round(x)-x) != 0.0)
-            throw AmiException("Invalid non-integer value for option in udata");
-        return(static_cast<int>(x));
+
+/** conversion from double to int with checking for loss of data
+  *  @param x input
+  *  @return int_x casted value
+  */
+int dbl2int(const double x){
+    if((std::round(x)-x) != 0.0)
+        throw AmiException("Invalid non-integer value for integer option");
+    return(static_cast<int>(x));
+}
+
+void setSolverOptions(const mxArray *prhs[], int nrhs, Solver &solver)
+{
+    if (!mxGetPr(prhs[RHS_OPTIONS])) {
+        throw AmiException("No options provided!");
```

Comment:
I think we can simply remove all the throwing of AmiExceptions here. We have default values anyways.

Reply:
still need the checks if the fields exists though.

Reply:
Would make code cleaner, agreed. A check for additional unused arguments might be more helpful. However, wouldn't do that right now.

Reply:
I don't think there is a good way to do this with the current mex interface.

Reply:
Just checking `nrhs` would be good enough.

Reply:
aah, true! I thought you were planning to check for additional fields in `prhs[RHS_OPTIONS]`.

Reply:
For those guys it should still be possible to iterate over field names and check for extraneous ones. Let's keep that for another time though. If we make more fields optional, we can also clean up the simulate_$.m

Reply:
agreed. Ideally simulate_$.m should do no checks at all.